pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1302,6 +1302,17 @@ self: super: {
     doHaddock = false;
     version = "2.0.7";
   });
+  hasura-ekg-core = super.hasura-ekg-core.overrideScope (self: super: {
+    hspec = dontCheck self.hspec_2_8_3;
+    hspec-core = dontCheck self.hspec-core_2_8_3;
+    hspec-discover = dontCheck super.hspec-discover_2_8_3;
+  });
+  hasura-ekg-json = super.hasura-ekg-json.overrideScope (self: super: {
+    ekg-core = self.hasura-ekg-core;
+    hspec = dontCheck self.hspec_2_8_3;
+    hspec-core = dontCheck self.hspec-core_2_8_3;
+    hspec-discover = dontCheck super.hspec-discover_2_8_3;
+  });
```

Comment:
It is somewhat annoying to have to basically add these same overrides to each package.

Here are two possible solutions, although I think they should be done in a separate PR:

1. Do not have `hasura-ekg-core` or `hasura-ekg-json` as top-level packages in `haskellPackages`, but instead just do something like:

    ```diff
    diff --git a/pkgs/development/haskell-modules/non-hackage-packages.nix b/pkgs/development/haskell-modules/non-hackage-packages.nix
    index 4aead5d1880..565be624177 100644
    --- a/pkgs/development/haskell-modules/non-hackage-packages.nix
    +++ b/pkgs/development/haskell-modules/non-hackage-packages.nix
    @@ -28,13 +28,14 @@ self: super: {
 
       # hasura graphql-engine is not released to hackage.
       # https://github.com/hasura/graphql-engine/issues/7391
    -  ci-info = self.callPackage ../misc/haskell/hasura/ci-info.nix {};
    -  pg-client = self.callPackage ../misc/haskell/hasura/pg-client.nix {};
    -  graphql-parser = self.callPackage ../misc/haskell/hasura/graphql-parser.nix {};
    -  graphql-engine = self.callPackage ../misc/haskell/hasura/graphql-engine.nix {};
    -  hasura-resource-pool = self.callPackage ../misc/haskell/hasura/pool.nix {};
    -  hasura-ekg-core = self.callPackage ../misc/haskell/hasura/ekg-core.nix {};
    -  hasura-ekg-json = self.callPackage ../misc/haskell/hasura/ekg-json.nix {};
    +  graphql-engine = self.callPackage ../misc/haskell/hasura/graphql-engine.nix {
    +    ci-info = self.callPackage ../misc/haskell/hasura/ci-info.nix {};
    +    ekg-core = self.callPackage ../misc/haskell/hasura/ekg-core.nix {};
    +    ekg-json = self.callPackage ../misc/haskell/hasura/ekg-json.nix {};
    +    graphql-parser = self.callPackage ../misc/haskell/hasura/graphql-parser.nix {};
    +    pg-client = self.callPackage ../misc/haskell/hasura/pg-client.nix {};
    +    resource-pool = self.callPackage ../misc/haskell/hasura/pool.nix {};
    +  };
    ```

    The problem with something like this is that deps like `resource-pool` would then not be easily overrideable in `configuration-common.nix`.

2. Create an overridescope argument and share it between all relevant packages:

    ```diff
    diff --git a/pkgs/development/haskell-modules/configuration-common.nix b/pkgs/development/haskell-modules/configuration-common.nix
    index e39efa48b30..770c28d20df 100644
    --- a/pkgs/development/haskell-modules/configuration-common.nix
    +++ b/pkgs/development/haskell-modules/configuration-common.nix
    @@ -1287,29 +1287,6 @@ self: super: {
       # Fails with "supports custom headers"
       Spock-core = dontCheck super.Spock-core;
     
    -  # hasura packages need some extra care
    -  graphql-engine = overrideCabal (super.graphql-engine.overrideScope (self: super: {
    -    immortal = self.immortal_0_2_2_1;
    -    resource-pool = self.hasura-resource-pool;
    -    ekg-core = self.hasura-ekg-core;
    -    ekg-json = self.hasura-ekg-json;
    -    hspec = dontCheck self.hspec_2_8_3;
    -    hspec-core = dontCheck self.hspec-core_2_8_3;
    -    hspec-discover = dontCheck super.hspec-discover_2_8_3;
    -    tasty-hspec = self.tasty-hspec_1_2;
    -  })) (drv: {
    -    patches = [ ./patches/graphql-engine-mapkeys.patch ];
    -    doHaddock = false;
    -    version = "2.0.7";
    -  });
    -  pg-client = overrideCabal (super.pg-client.override {
    -    resource-pool = self.hasura-resource-pool;
    -  }) (drv: {
    -    librarySystemDepends = with pkgs; [ postgresql krb5.dev openssl.dev ];
    -    # wants a running DB to check against
    -    doCheck = false;
    -  });
    -
       # https://github.com/bos/statistics/issues/170
       statistics = dontCheck super.statistics;
     
    @@ -1948,3 +1925,31 @@ EOT
       streamly-posix = doJailbreak super.streamly-posix;
     
     } // import ./configuration-tensorflow.nix {inherit pkgs haskellLib;} self super
    +//
    +  # hasura packages needs some extra overrides
    +  (let
    +      scope = self: super: {
    +        immortal = self.immortal_0_2_2_1;
    +        resource-pool = self.hasura-resource-pool;
    +        ekg-core = self.hasura-ekg-core;
    +        ekg-json = self.hasura-ekg-json;
    +        hspec = dontCheck self.hspec_2_8_3;
    +        hspec-core = dontCheck self.hspec-core_2_8_3;
    +        hspec-discover = dontCheck super.hspec-discover_2_8_3;
    +        tasty-hspec = self.tasty-hspec_1_2;
    +      };
    +   in {
    +     graphql-engine = overrideCabal (super.graphql-engine.overrideScope scope) (drv: {
    +       patches = [ ./patches/graphql-engine-mapkeys.patch ];
    +       doHaddock = false;
    +       version = "2.0.7";
    +     });
    +     hasura-ekg-core = super.hasura-ekg-core.overrideScope scope;
    +     hasura-ekg-json = super.hasura-ekg-json.overrideScope scope;
    +     pg-client = overrideCabal (super.pg-client.overrideScope scope) (drv: {
    +       librarySystemDepends = with pkgs; [ postgresql krb5.dev openssl.dev ];
    +       # wants a running DB to check against
    +       doCheck = false;
    +     });
    +   }
    +  )
    ```

-------------------------------

@Lassulus you may want to make one of these changes in a future update to hasura in order to make maintenance easier.

Reply:
I like the second solution. I have done similar things with hls.
Only issue with this is that we loose the duplicate key errors if we end up with overlays in both attrsets.

Reply:
I would have taken the first solution. Since I don't think we need to export any of the dependencies?

Reply:
Yeah, probably smarter in this case.